lib/IDE/CodeCompletion.cpp

```diff
@@ -6111,12 +6093,48 @@ bool CodeCompletionCallbacksImpl::trySolverCompletion() {
   }
 }
 
+// FIXME: Remove this once all expression position completions are migrated
+// to work via TypeCheckCompletionCallback.
+static void undoSingleExpressionReturn(DeclContext *DC) {
+  auto updateBody = [](BraceStmt *BS, ASTContext &Ctx) -> bool {
+    if (Ctx.CompletionCallback)
+      return false;
+
+    SourceManager &SM = Ctx.SourceMgr;
+    ASTNode FirstElem = BS->getFirstElement();
+    auto *RS = dyn_cast_or_null<ReturnStmt>(FirstElem.dyn_cast<Stmt*>());
+
+    if (!RS || !RS->isImplicit())
+      return false;
+
+    SourceRange Range = RS->getSourceRange();
+    if (Range.isInvalid())
+      return false;
+
+    if (!SM.rangeContainsCodeCompletionLoc(Range))
+      return false;
```

Comment:
It's not clear why we need to compare the range. We know completion is happening inside this `DC`, right? Are these `return false` actually reachable? If so, could you give me some examples?

Also, if we really need to check it, `RS->getResult()->getSourceRange()` is more relevant I think.